### ISSUE TITLE
[Feature Request]: Add the "Output steps metrics" transform to HOP

### ISSUE THREAD

**Issue by jayyellow86** [apache/hop#3003]
### What would you like to happen?

In PDI/Kettle, there was a Transformation Step called "Output steps metrics", which is missing in HOP transforms.
This step is configured with a list of steps from the same Transformation and the field names for a fixed set of metrics.
During execution this step will output these metrics after the listed steps are finished.
Please add this step as a transform to HOP.

### Issue Priority

Priority: 3

### Issue Component

Component: Transforms

**Comment by mattcasters** [comment]
Workaround is to get this information after the execution with the "Execution Information" transform in another pipeline.

**Comment by openinbigdata** [comment]
It would be fantastic to have this functionality, just yesterday I needed it

**Comment by openinbigdata** [comment]
> Workaround is to get this information after the execution with the "Execution Information" transform in another pipeline.

Matt, but is reading the "Execution Information" information already possible in 2.7?

**Comment by mattcasters** [comment]
> > Workaround is to get this information after the execution with the "Execution Information" transform in another pipeline.
> 
> Matt, but is reading the "Execution Information" information already possible in 2.7?

Yes, this is already possible.  There's a whole perspective to explore prior executions, but this information is also available through the transform.

**Comment by openinbigdata** [comment]
> > > Workaround is to get this information after the execution with the "Execution Information" transform in another pipeline.
> > 
> > 
> > Matt, but is reading the "Execution Information" information already possible in 2.7?
> 
> Yes, this is already possible. There's a whole perspective to explore prior executions, but this information is also available through the transform.

Matt, do you have any example pipelines using the Execution information transform that you can send me? I confess that I can't use it, as I don't understand how it should be used, so an example would be of great help.

**Comment by 172151295** [comment]
还是希望实现转换步骤统计这个组件一样的功能？目前hop没看到有类似功能的组件，实现检测插入更新组件的插入、更新、读取、等

**Comment by 172151295** [comment]
转换步骤统计信息这个组件没有！！

**Comment by bamaer** [comment]
did you look at the [Execution Information perspective](https://hop.apache.org/manual/latest/hop-gui/perspective-execution-information.html) and [Execution Information transform](https://hop.apache.org/manual/latest/pipeline/transforms/execinfo.html)?

**Comment by krishnasdfw** [comment]
I reviewed Exec Info Transform. It seems to read from audit location, but not clear whether that can be directly tied to a current running transform and it's data.  Pipeline Log / Pipeline Probe combination seems like an option, but can both the log and probe be setup in same pipeline and both meta data types configured with same pipeline? Will that be a single execution of the pipeline feeding into both inputs in same pipeline execution? This is a valid scenario if the input is just one row of large json message that may get split rows processed during pipeline execution. Ultimately the error handler that examines error hop step metrics , may need to get this first input message and take some action based on error metrics.

**Comment by CarlosJuncher03** [comment]
Regarding this request, do you plan to include it? I think it would be good, as it would make it easier to add information about pipeline totals using the write-to-log feature. The write-to-log feature makes it easy to put important information into the standard pipeline log in a simple way.

**Comment by mattcasters** [comment]
It would be better to explain what you exactly need as I have no idea anymore about what the old step exactly did. Do you wonderful folks really think it's a good idea to manually add a bunch of extra transforms to each pipeline despite Pipeline Logging offering the exact same information in a generic way?  Simply go to the metadata perspective, create a new Pipeline Logging element, create a new logging pipeline and you'll get a generic solution that works on your whole project.

**Comment by CarlosJuncher03** [comment]
I agree that Pipeline Logging is the most suitable generic solution, and we already use Pipeline Logging along with Pipeline Probe for monitoring and controlling the data path.

The use case I see here is different: small projects or restricted environments where the standard pipeline log is the only practical place to expose metrics. For example, a simple pipeline running on third-party infrastructure, with limited server access, where configuring a full logging pipeline or centralized monitoring is impractical.

In this situation, having a lightweight way to write selected metrics from the steps directly to the standard log would be useful. The goal wouldn't be to replace Pipeline Logging, but to make simple cases easier and more visible.

Perhaps this could be implemented as an improvement to Write to Log, or as a simplified way to record selected metrics without creating a separate logging pipeline.

**Comment by CarlosJuncher03** [comment]
I agree that Pipeline Logging, Pipeline Probe, and Execution Information are excellent ways to monitor the full pipeline flow in a generic and reusable way.

In fact, we have been able to create a monitoring model that can be forked and applied to multiple pipelines, which is very powerful.

My point is not to replace that approach. The idea is to have something lighter for smaller or more niche pipelines, where setting up the full logging/probe/execution information structure may be more than what is needed.

For those cases, it would be useful to have a simple way to expose selected step metrics directly in the standard pipeline log, or through a lightweight transform. This would make basic operational visibility easier without competing with the more complete monitoring approach.

**Comment by CarlosJuncher03** [comment]
Of course, this is just my perspective based on some use cases we've seen. I understand that there might be better ways to solve this within Hop's current architecture, but I believe something along these lines could add value.

**Comment by mattcasters** [comment]
Created by my friend Samatar, that's why it escaped my attention:  https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/main/java/org/pentaho/di/trans/steps/stepsmetrics/StepsMetrics.java
